src/main/java/com/iota/iri/cache/impl/CacheImpl.java

```diff
@@ -76,47 +74,58 @@ public int getSize() {
 
     @Override
     public void put(K key, V value) {
-        if (getSize() == cacheConfiguration.getMaxSize()) {
-            evict();
-        }
         // new entry
         if (strongStore.put(key, value) == null) {
-            evictionQueue.offer(key);
+            releaseQueue.offer(key);
         }
     }
 
     @Override
-    public void evict(K key) {
+    public void release(K key) {
         if (key == null || !strongStore.containsKey(key)) {
             return;
```

Comment:
When this fires, it should remove the key from the release queue so as to not continually release this. When the `TransactionViewModel.fromHash()` function is called it triggers a cache release, but the release does not update the queue when returning null here.

Reply:
Ah yes a not on this i didn't mention above, for this release deletion to be effective you should revert the last change you made in commit https://github.com/iotaledger/iri/pull/1718/commits/43ad3ba064b7b277fd25f49eb07906d3149201dc

Reply:
did you act on this?

Reply:
Yeah.

Reply:
```
01/23 13:15:16.757 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27744
01/23 13:15:26.758 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27744
01/23 13:15:36.759 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27744
01/23 13:15:51.760 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27744
01/23 13:16:01.761 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27744
01/23 13:16:11.762 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27744
01/23 13:16:26.763 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27744
01/23 13:16:36.764 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27744
01/23 13:16:46.765 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27744
01/23 13:16:56.766 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27744
01/23 13:17:06.767 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27744
01/23 13:17:16.768 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27744
01/23 13:17:26.770 [Tips Requester] INFO  TipsRequesterImpl:86 - toProcess = 0 , toBroadcast = 0 , toRequest = 0 , toReply = 0 / totalTransactions = 27744
```
This is what I like to see!